src/device.rs

```diff
@@ -409,8 +436,8 @@ pub trait DeviceV1_0 {
                                        pipelines.as_mut_ptr());
         pipelines.set_len(create_infos.len());
         match err_code {
-            vk::Result::Success => Ok(pipelines),
-            _ => Err(err_code),
+            vk::Result::Success => (pipelines, Ok(())),
+            _ => (pipelines, Err(err_code)),
```

Comment:
I think i might slightly adjust the return type to `Result<Vec<vk::Pipeline>, (Vec<vk::Pipeline>, vk::Result)>`.  This guarantees that on success that there will not be any null handles inside the vec, and you can still recover if you need too. Any objections?